packages/coil-extension/docs/iframes/iframes.md

```diff
@@ -0,0 +1,334 @@
+- **chrome.runtime.sendMessage()**: This method allows content scripts to send messages to the background script.
+- **chrome.tabs.sendMessage()**: This method can be used to send messages from the background script to a specific content script running in a frame identified by a tabId and frameId (or all frames).
+- **chrome.runtime.onMessage**: This event can be used to listen for messages sent between components.
+- **window.postMessage()**: This method can be used to send messages between windows (or iframes) in a secure way. It can be used to communicate between the initiatorContentScript and the parentContentScript.
+- **iframe.contentWindow**: This property of an `<iframe>` element provides access to the Window object of the document inside the iframe and can be used to send messages between the parent and child documents using the postMessage() method.
+- **window.parent**: This property in the initiatorContentScript can be used to reference the parent window (the window containing the iframe).
+- **event.source**: When handling a message event, this property can be used to get the Window object of the message sender. It can be used in the parentContentScript to compare the source of the message with the contentWindow property of each iframe.
+
+### Security Concerns:
+
+To prevent potential security vulnerabilities, message payloads sent between frames using postMessage() should only include correlationIds and not include other information, such as frameId or tabId. This is because correlationIds can be generated randomly and do not reveal any sensitive information about the frame or tab. If other information is included, it could potentially reveal sensitive information and open up the possibility for eavesdropping.
+
+It's possible to do matching of event.source (of 'message' event) against iframe.contentWindow
+to do correlation though it's unclear how that could be done with out sending the frameId/tabId.
+
+### Solution overview
+
+The following is how iframe monetization is implemented. It's possible there's a simpler way which
+doesn't need the webNavigation api (maybe using window.parent.postMessage to get frame ancestry),
+though if you are requesting <all_urls> the permission UX is largely unchanged.
+
+1. The initiatorContentScript sends a request to the backgroundScript to check if monetization is allowed for the iframe.
+2. The backgroundScript determines the parent frame and sends a request to the parentContentScript to verify if the iframe's monetization is allowed based on its attributes.
+3. The parentContentScript generates a unique correlationId for each unknown iframe, sends it to the corresponding content script using window.postMessage(), and awaits a response. Note that the message might be sent to a random iframe's content script and not necessarily the initiatorContentScript.
```

Comment:
TODO: parent script does not await a response here

Reply:
well not in the sense of a direct response to a message ala chrome.runtime.sendMessage(message, responseCallback)